cli/flox-rust-sdk/src/models/manifest/composite/shallow.rs

```diff
@@ -0,0 +1,499 @@
+        if low_priority != high_priority {
+            unreachable!("versions are hardcoded into Manifest");
+        }
+
+        Ok(high_priority.clone())
+    }
+
+    fn merge_install(
+        low_priority: &ManifestInstall,
+        high_priority: &ManifestInstall,
+    ) -> Result<ManifestInstall, MergeError> {
+        let merged = map_union(low_priority.inner(), high_priority.inner());
+        Ok(ManifestInstall(merged))
+    }
+
+    /// Keys in `manifest2` overwrite keys in `manifest1`.
+    fn merge_vars(
+        low_priority: &ManifestVariables,
+        high_priority: &ManifestVariables,
+    ) -> Result<ManifestVariables, MergeError> {
+        let merged = map_union(low_priority.inner(), high_priority.inner());
+        Ok(ManifestVariables(merged))
+    }
+
+    fn merge_hook(
+        low_priority: &ManifestHook,
+        high_priority: &ManifestHook,
+    ) -> Result<ManifestHook, MergeError> {
+        Ok(ManifestHook {
+            on_activate: append_optional_strings(
+                low_priority.on_activate.as_ref(),
+                high_priority.on_activate.as_ref(),
+            ),
+        })
+    }
+
+    fn merge_profile(
+        low_priority: &ManifestProfile,
+        high_priority: &ManifestProfile,
+    ) -> Result<ManifestProfile, MergeError> {
+        let common =
+            append_optional_strings(low_priority.common.as_ref(), high_priority.common.as_ref());
+        let bash = append_optional_strings(low_priority.bash.as_ref(), high_priority.bash.as_ref());
+        let zsh = append_optional_strings(low_priority.zsh.as_ref(), high_priority.zsh.as_ref());
+        let tcsh = append_optional_strings(low_priority.tcsh.as_ref(), high_priority.tcsh.as_ref());
+        let fish = append_optional_strings(low_priority.fish.as_ref(), high_priority.fish.as_ref());
+        let merged = ManifestProfile {
+            common,
+            bash,
+            zsh,
+            fish,
+            tcsh,
+        };
+        Ok(merged)
+    }
+
+    fn merge_options(
+        low_priority: &ManifestOptions,
+        high_priority: &ManifestOptions,
+    ) -> Result<ManifestOptions, MergeError> {
+        let mut merged = low_priority.clone();
+        merged.allow.unfree = high_priority.allow.unfree.or(low_priority.allow.unfree);
+        merged.allow.broken = high_priority.allow.broken.or(low_priority.allow.broken);
+        merged.allow.licenses = if high_priority.allow.licenses.is_empty() {
+            low_priority.allow.licenses.clone()
+        } else {
+            high_priority.allow.licenses.clone()
+        };
+        merged.semver.allow_pre_releases = high_priority
+            .semver
+            .allow_pre_releases
+            .or(low_priority.semver.allow_pre_releases);
+        merged.cuda_detection = high_priority.cuda_detection.or(low_priority.cuda_detection);
+        merged.systems = high_priority
+            .systems
+            .clone()
+            .or(low_priority.systems.clone());
```

Comment:
note nonblocking: some of these methods might be less boilerplatey if we had a `deep_merge()` function that took something that could be converted to a `serde_json::Value` and merged that

Reply:
That gets tricky with some of the options. Does a deep merge append arrays? You probably don't want to append the arguments in `cmd` for example. That's one of the reasons I just did the fields individually, it's clear how each field is merged and you aren't forced to do an all-or-nothing merge strategy.